app/controllers/users_controller.rb

```diff
@@ -0,0 +1,120 @@
+class UsersController < ApplicationController
+  
+  # Whitelist
+  skip_before_action :logged_in_user, except: [:edit, :update, :index, :destroy]
+  skip_before_action :correct_user, except: [:edit, :update, :destroy]
+  before_action :set_user, except: [:index, :new, :create]
+
```

Comment:
would be helpful to add a after_action sign_in, only: [:create] or something like that so that the user is logged in after he/she makes an account